addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/handler/EdimaxHandler.java

```diff
@@ -0,0 +1,130 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.edimax.EdimaxBindingConstants;
+import org.openhab.binding.edimax.configuration.EdimaxConfiguration;
+import org.openhab.binding.edimax.internal.ConnectionInformation;
+import org.openhab.binding.edimax.internal.commands.GetState;
+import org.openhab.binding.edimax.internal.commands.SetState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EdimaxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Falk Harnisch - Initial contribution
+ */
+public class EdimaxHandler extends BaseThingHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EdimaxHandler.class);
+
+    private static final int PORT = 10000;
+
+    /**
+     * Connection Information to the device.
+     */
+    protected ConnectionInformation ci;
+
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Constructor for the edimax things.
+     *
+     * @param thing The thing
+     */
+    public EdimaxHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        LOGGER.debug("command for {}: {}", channelUID.getAsString(), command.toString());
+        try {
+            if (channelUID.getId().equals(EdimaxBindingConstants.SWITCH)) {
+                if (command instanceof RefreshType) {
+                    LOGGER.debug("State: {}", getState().toString());
+                    if (getState()) {
+                        updateState(channelUID, OnOffType.ON);
+                    } else {
+                        updateState(channelUID, OnOffType.OFF);
+                    }
+                }
+                if (command instanceof OnOffType) {
+                    if (command.equals(OnOffType.ON)) {
+                        switchState(Boolean.TRUE);
+                    } else {
+                        switchState(Boolean.FALSE);
+                    }
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        LOGGER.debug("Initializing Things");
+        final EdimaxConfiguration configuration = this.getConfigAs(EdimaxConfiguration.class);
+        final String completeUrl = completeURL(configuration.getIpAddress());
+        ci = new ConnectionInformation(configuration.getUsername(), configuration.getPassword(), completeUrl, PORT);
```

Comment:
What is the (real) difference between the `EdimaxConfiguration` and the `ConnectionInformation` do you really need them both? Imho you don't need both of them, any reasons you think you do?